## <p style="text-align: right;"> &#9989; Put your name here</p>

# CMSE 201: Quiz 1 (Section 002 - Spring 2022)

The goal of this quiz is to give you the opportunity to test out some of the skills that you've developed thus far this semester. In particular, you will reflect on what you know about writing code using Python and showcase some of your new programming skills.

**Important note about using online resources**: This exam is "open internet". That means that you can look up documentation, google how to accomplish certain Python tasks, etc. Being able to effectively use the internet for computational modeling and data science is a very important skill, so we want to make sure you have the opportunity to exercise that skill. **However, the use of any person-to-person communication software is absolutely not acceptable.**

**Do your own work.** This quiz is designed to give *you* the opportunity to show the instructor what you can do and you should hold yourself accountable for maintaining a high level of academic integrity. Any violation of academic integrity could result in you receiving a zero on the quiz.

---
# Academic integrity statement (0.5 point)

In the markdown cell below, put your personal academic integrity statement. By including this statement, you are confirming that you are submitting this as your own work and not that of someone else.


<font size=+3>&#9998;</font> *Put your personal academic integrity statement here.*

---

# Calculating Catering Food Costs  (8 points)

You are helping a catering company to figure out how much they need to spend on ordering materials. For each catering job, the system stores three lists:
- One with the name of the item 
- One with the quantity of that item they need 
- One with the price of the item 

**Your company needs a piece of code that will calculate some relevant information on the cost for each job, so they can make up the budget.** 

In [1]:
materials_names = ["Eggplant", "Roast Beef", "Eggs", "Strawberries", "Pasta", "Parmesan Cheese"]
materials_quantity = [100, 50, 500, 2500, 100, 10]
materials_prices = [2.5, 25.0, 0.1, 0.20, 12, 30]

&#9989;&nbsp; **Part 1 (3 points):** To calculate the total cost *for one item*, the item's price must be multiplied by the quantity needed (E.g., the total cost of `Eggplants` is \\$2.50 per eggplant $\times$ 100 eggplants $=$ \$250.00.) The total cost of materials is the sum of each item's total cost. Your task is to compute the **total cost of materials** using a **for loop**, and **print** the result in the following format _The total cost is $\$$ ...__ 

Below, there are two cells. In the first one, you will put your **pseudo-code** (variables, structure, equations, iteration variable, ...). In the second one, you will put your **code**.

<font size=6 color="#009600">&#9998;</font>  put your **pseudo-code** here!

### ANSWER ###

If it makes sense **(1 Point)** 

In [2]:
# Put your code here! 
total_cost = 0                                                  
for jj in range(len(materials_quantity)):                       
    total_cost += materials_quantity[jj]*materials_prices[jj]   
print("The total cost is $", total_cost)                        

The total cost is $ 3550.0


&#9989;&nbsp; **Part 2 (4 points):** To have a better idea of the budget, your boss wants to know the percentage of the total cost associated with each item (E.g., the percentage associated with Eggplants is (total cost of Eggplants / total cost of materials) $\times$ 100 $\%$). Create a **list** called `percentage_item`. Using a **while loop**, fill this list with the percentage of the total cost associated with each item. **Print** the list `percentage_item`.

Below, there are two cells. In the first one, you will put your **pseudo-code** (variables, structure, equations, iteration variable, ...). In the second one, you will put your **code**.

**Note** that if you did not succeed to compute the total cost, please use the following value `total_cost = 3550` 

<font size=6 color="#009600">&#9998;</font>  put your **pseudo-code** here!

### ANSWER ###

If it makes sense **(1 Point)** 

In [3]:
# Put your code here!
percentage_item = list()                                                                
jj = 0                                                                                   
while jj < len(materials_quantity):                                                      
    percentage_item.append(materials_quantity[jj]*materials_prices[jj]/total_cost*100)   
    jj += 1                                                                              
print(percentage_item)                                                                   

[7.042253521126761, 35.2112676056338, 1.4084507042253522, 14.084507042253522, 33.80281690140845, 8.450704225352112]


&#9989;&nbsp; **Part 3 (1 point):** Now, you want to verify your code to be sure that it does what you want. Create a simple verification test for your code in **Part 2**.

Below, there are two cells. In the first one, you will put your **pseudo-code** (variables, structure, iteration variable, ...). In the second one, you will put your **code**.

<font size=6 color="#009600">&#9998;</font>  put your **pseudo-code** here!

### ANSWER ###

If it makes sense **(0.5 Point)** 

In [4]:
# Put your code here!
print(sum(percentage_item))              

99.99999999999999


---

### Congratulations, you're done!

Submit this quiz by uploading it to the course Desire2Learn web page.  Go to the "Quizzes" section, find the submission folder link for Quiz #1, and upload it there.